contracts/Zauction.sol

```diff
@@ -181,4 +255,14 @@ contract ZAuction is Initializable, OwnableUpgradeable {
   function toEthSignedMessageHash(bytes32 hash) public pure returns (bytes32) {
     return hash.toEthSignedMessageHash();
   }
+
+  function rootDomainIdOf(uint256 id) public view returns (uint256) {
+    uint256 parentId = registrar.parentOf(id);
+    uint256 holder = id;
```

Comment:
Something interesting about how this code will work;

If someone sells a top level domain which has a fee, then the top level domain fee applies to the top level domain itself. This isn't necessarily bad, but worth pointing out somewhere.